vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionManager.java

```diff
@@ -66,4 +79,42 @@ public IExhibition getStartExhibition() {
         }
         return null;
     }
+
+    /**
+     * Updates the Exhibition with given list of languages. It fetches the language from exhibitionLanguageConfig using code.
+     *  
+     * @param exhibition
+     * @param defaultLanguage 
+     * @param languages
+     */
+    @Override
+    public void updateExhibitionLanguages(Exhibition exhibition, List<String> codes, String defaultLanguage) {
+        if(CollectionUtils.isNotEmpty(codes)) {
+            if(defaultLanguage!=null) {
+                codes.add(defaultLanguage);
+            }
+            if(CollectionUtils.isNotEmpty(exhibitionLanguageConfig.getExhibitionLanguageList())) {
+                List<ExhibitionLanguage> exhibitionLanguages =   exhibitionLanguageConfig.getExhibitionLanguageList()
+                        .stream().filter(languageMap -> codes.contains(languageMap.get(ConfigConstants.CODE)))
+                        .map(map -> {
+                            ExhibitionLanguage exhibitionLanguage =   new ExhibitionLanguage((String) map.get(ConfigConstants.LABEL),
+                                    (String) map.get(ConfigConstants.CODE), exhibition);
+
+                            if(exhibitionLanguage.getCode().equalsIgnoreCase(defaultLanguage)) {
+                                exhibitionLanguage.setDefault(true);
+                            }
+                            return exhibitionLanguage;
+                        }).collect(Collectors.toList());
+
+                if(CollectionUtils.isNotEmpty(exhibitionLanguages)) {
+                    logger.info("Updating Exhibition with Languages" + codes);
+                    exhibition.getLanguages().addAll(exhibitionLanguages);
+                }
+            } else {
+                logger.error("Language list configuration not available");
```

Comment:
in this case we probably want to throw an exception since that would indicate a faulty installation.